internal/ad/download.go

```diff
@@ -196,22 +192,26 @@ func needsDownload(ctx context.Context, client *libsmbclient.Client, g *download
 	return true, nil
 }
 
-func getGPOVersion(r io.Reader) (version int, err error) {
+func getGPOVersion(ctx context.Context, r interface{}, downloadableName string) (version int, err error) {
```

Comment:
I’m not a big fan of changing this to an interface{} to support both filename or reader. It means that we don’t control the contract of what we accept in our function.

My suggestion would be then to always take a reader, readall the file in that function and call Load with the slice of bytes, which, reading from the source code, it seems to accept. That way, we don’t leak the lack of contract that ini.Load() imposes on us. wdyt?